src/app/checkpoints/Checkpoints.tsx

```diff
@@ -1,53 +1,110 @@
-import Image from "next/image"
-import { Card } from "~/components/Card"
 import { ErrorCard } from "~/components/ErrorCard"
-import { Container } from "~/components/layout/Container"
 import { Flex } from "~/components/layout/Flex"
 import { Grid } from "~/components/layout/Grid"
-import { H4 } from "~/components/typography/H4"
 import { getCheckpoints } from "~/services/d2checkpoint/client"
 import { prefetchManifest } from "~/services/raidhub/prefetchRaidHubManifest"
+import { theme } from "../layout/theme"
 import { ClientControls } from "./ClientControls"
 import { CopyButton } from "./CopyButton"
+import { CheckpointCard } from "./cards/CheckpointCard"
+import { CheckpointCardContentSection } from "./cards/CheckpointCardContentSection"
+
+type Checkpoint = {
+    encounterImageURL: string
+    maxPlayers: number
+    currentPlayers: number
+    encounterName: string
+    difficultyTier: string
+    botBungieName: string
+    activityHash: string
+    activityName: string
+}
+
+type GroupedCheckpoint = {
+    title: string
+    checkpoints: Checkpoint[]
+}
 
 export const Checkpoints = async () => {
     const [checkpoints, manifest] = await Promise.all([getCheckpoints(), prefetchManifest()])
 
+    const groupedCheckpoints: GroupedCheckpoint[] = checkpoints.checkpoints.reduce(
+        (groups: GroupedCheckpoint[], checkpoint: Checkpoint) => {
+            const existingGroup = groups.find(
+                (group: GroupedCheckpoint) =>
+                    group.title ===
+                    checkpoint.activityName
+                        .replaceAll(`${checkpoint.difficultyTier}`, "")
+                        .replaceAll(":", "")
+            )
+            if (existingGroup) {
+                existingGroup.checkpoints.push(checkpoint)
+            } else {
+                groups.push({
+                    title: checkpoint.activityName
+                        .replaceAll(`${checkpoint.difficultyTier}`, "")
+                        .replaceAll(":", ""),
+                    checkpoints: [checkpoint]
+                })
+            }
+            return groups
+        },
+        []
+    )
+
     return (
-        <Flex $padding={0} $direction="column" $crossAxis="flex-start" $fullWidth>
-            {checkpoints.alert.active && <ErrorCard>{checkpoints.alert.message}</ErrorCard>}
-            <ClientControls date={checkpoints.updatedAt} />
-            <Grid $fullWidth>
-                {checkpoints.checkpoints
-                    .filter(checkpoint =>
-                        Object.keys(manifest.hashes).includes(checkpoint.activityHash)
+        <>
```

Comment:
A rule of thumb I like to go with is to try and make all exported components export a single component, i.e, no react fragments. Makes it so the component exported is not dependent on the parent. 

Sometimes this can be fixed moving the parent logic into this component, or sometimes you'll need to wrap the component with something else.